vyper/codegen/core.py

```diff
@@ -165,13 +164,13 @@ def _dynarray_make_setter(dst, src, pos=None):
 
                 return b1.resolve(b2.resolve(["seq", store_len, loop]))
 
-        if src.value is None:
+        if src.value == "~empty":
```

Comment:
The `~empty` magic value seems kind of magical. At least make it a constant.

Probably need a better DSL object system anyways.

Reply:
We use the same pattern with a lot of other macros like `~calldata`, `~codelen`. I considered some more object oriented approaches, but.. we don't really have a design/refactor plan for that so I opted for more incremental improvement